### PR TITLE
add 2.5 release notes

### PR DESCRIPTION
**Pull request by YANG-DB**
Signed-off-by: YANGDB <yang.db.dev@gmail.com>

### Description
add 2.5 release notes
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-sql.release-notes-2.5.0.0.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+Compatible with OpenSearch and OpenSearch Dashboards Version 2.5.0
+
+### Features
+
+#### Structure changes
```
Comment (dai-chen) [inline release-notes/opensearch-sql.release-notes-2.5.0.0.md:5]: Please pay attention to the category naming: https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md#categories. Core team needs to map each category here to a consolidated release notes finally.

File: release-notes/opensearch-sql.release-notes-2.5.0.0.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+#### Security Related Enhancements
+
+* Add security patch for CVE-2020-15250 ([#1095](https://github.com/opensearch-project/sql/issues/1095))
+* Add security patch for CVE-2022-45868 ([#1107](https://github.com/opensearch-project/sql/issues/1107))
+
+### Bug Fixes
+
+* Add missing API method in prometheus table
+* add table default impl of exist and create
+* Fix `FLOAT` -&gt; `DOUBLE` cast. ([#1025](https://github.com/opensearch-project/sql/issues/1025))
```
Comment (Yury-Fridlyand) [inline release-notes/opensearch-sql.release-notes-2.5.0.0.md:62]: Should be?
```suggestion
* Fix `FLOAT` -> `DOUBLE` cast. ([#1025](https://github.com/opensearch-project/sql/issues/1025))
```

File: release-notes/opensearch-sql.release-notes-2.5.0.0.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+* Fix back quoted alias of FROM subquery ([#1189](https://github.com/opensearch-project/sql/issues/1189) [#1208](https://github.com/opensearch-project/sql/issues/1208))
+* Fix truncate function ([#1197](https://github.com/opensearch-project/sql/issues/1197) [#1213](https://github.com/opensearch-project/sql/issues/1213))
+* Allow common keywords and scalar function name used as identifier ([#1191](https://github.com/opensearch-project/sql/issues/1191) [#1212](https://github.com/opensearch-project/sql/issues/1212))
+* Suppress report uploading failure in CI. ([#1180](https://github.com/opensearch-project/sql/issues/1180) [#1220](https://github.com/opensearch-project/sql/issues/1220))
+* Fixed error with single timestamp query ([#1244](https://github.com/opensearch-project/sql/issues/1244) [#1246](https://github.com/opensearch-project/sql/issues/1246) [#1249](https://github.com/opensearch-project/sql/issues/1249))
+* Support JOIN query on object field with unexpanded name ([#1229](https://github.com/opensearch-project/sql/issues/1229) [#1250](https://github.com/opensearch-project/sql/issues/1250))
+
+
+
```
Comment (Yury-Fridlyand) [inline release-notes/opensearch-sql.release-notes-2.5.0.0.md:81]:
```suggestion
```

File: release-notes/opensearch-sql.release-notes-2.5.0.0.md (hunk 1 above)
Comment (YANG-DB) [reply release-notes/opensearch-sql.release-notes-2.5.0.0.md:5]: ok thanks...

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by vamsi-amazon — DISMISSED** [review]

**Review by penghuo — APPROVED** [review]

File: release-notes/opensearch-sql.release-notes-2.5.0.0.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+* Add Alternate Syntax For Match_Query And Other Functions ([#1166](https://github.com/opensearch-project/sql/issues/1166))
+* Add Support For `TIME` Type in DAY_OF_YEAR Functions ([#199](https://github.com/opensearch-project/sql/issues/199) [#1223](https://github.com/opensearch-project/sql/issues/1223) [1#258](https://github.com/opensearch-project/sql/issues/1258))
+* Add Day_Of_Month Function As An Alias Of DayOfMonth ([#1227](https://github.com/opensearch-project/sql/issues/1227) [#1265](https://github.com/opensearch-project/sql/issues/1265))
+* Add security patch for CVE-2020-15250 ([#1095](https://github.com/opensearch-project/sql/issues/1095))
+* Add security patch for CVE-2022-45868 ([#1107](https://github.com/opensearch-project/sql/issues/1107))
+
+### Bug Fixes
+
+* Add missing API method in prometheus table
+* add table default impl of exist and create
```
Comment (joshuali925) [inline release-notes/opensearch-sql.release-notes-2.5.0.0.md:59]: pr links?

**Review by vamsi-amazon — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Pull request merged**